src/app/router/Popup/Popup.tsx

```diff
@@ -56,12 +59,23 @@ function Popup() {
 }
 
 const Layout = () => {
+  const { account } = useAccount();
+  // keep last tries result, to prevent component flick during loading state
+  const [lastAccountFailed, setLastAccountFailed] = useState<boolean>(false);
+
+  useEffect(() => {
+    // skip the state during loading phase where only the id is present
+    if (Object.prototype.hasOwnProperty.call(account, "name")) {
+      setLastAccountFailed(!!account?.error);
+    }
+  }, [setLastAccountFailed, account]);
+
```

Comment:
That feels a bit complex on the first sight. What is it that you want to prevent here? 

Isn't it natural to go from error -> loading -> success / error?

Reply:
You can disable it and see it. If you replace 
`{lastAccountFailed || account?.error ? <ConnectionError /> : <Outlet />}` with 
`{account?.error ? <ConnectionError /> : <Outlet />}` you see during the loading state after clicking on "Try again" you see the `Outlet` component for a short moment. A glitch...